lib/streamlit/runtime/state/query_params.py

```diff
@@ -84,16 +89,19 @@ def __delattr__(self, key: str) -> None:
             raise AttributeError(_missing_key_error_message_query_params(key))
 
     def _delitem(self, key):
+        self.check_both_apis_used()
         if key in self._query_params:
             del self._query_params[key]
             self._send_query_param_msg()
         else:
             raise KeyError(_missing_key_error_message_query_params(key))
 
     def __contains__(self, key: str) -> bool:  # type: ignore[override]
+        self.check_both_apis_used()
         return key in self._query_params
 
     def __len__(self) -> int:
+        self.check_both_apis_used()
```

Comment:
not sure if we really need to add it to these methods that don't actually change state. I just added for now since it's NBD and probably better to be conservative but happy to remove.